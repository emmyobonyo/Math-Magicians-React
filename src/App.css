body {
  background-color: white;
  margin: auto;
}

.calculator {
  width: 50%;
  margin: auto;
  margin-top: 50px;
}

.calculator-screen {
  background-color: #858693;
  text-align: right;
}

.calculator-digits {
  font-size: 50px;
  padding: 5px 10px;
  font-weight: bold;
  margin: 0;
  color: white;
}

.calculator-buttons {
  display: grid;
  grid-template-columns: 25% 25% 25% 25%;
}

.calculator-button {
  background-color: #e0e0e0;
  border: 1px solid #cacaca;
  padding: 20px;
  font-size: 30px;
  text-align: center;
}

.orange-button {
  background-color: #f5913e;
}

.calculator-buttons-bottom {
  display: grid;
  grid-template-columns: 50% 25% 25%;
}

.calculator-button-bottom {
  background-color: #e0e0e0;
  border: 1px solid #cacaca;
  padding: 20px;
  font-size: 30px;
  text-align: center;
}

.orange-button-bottom {
  background-color: #f5913e;
}

.navigation {
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: green;
  padding: 0 30px;
}

.navigation-links {
  list-style-type: none;
  margin: 0;
  padding: 0;
  overflow: hidden;
}

.navigation-links li {
  float: left;
  margin-left: 20px;
}

a {
  text-decoration: none;
  color: white;
  font-weight: bold;
}

.logo {
  color: white;
}
