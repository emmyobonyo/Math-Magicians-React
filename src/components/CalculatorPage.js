import React from 'react';
import Calculator from './calculator';

function CalculatorPage() {
  return (
    <div className="calculator-page">
      <div>
        <h2 className="calculator-page-heading">Let us do some Math</h2>
        <p>These are the calculations that you can perform</p>
        <ul className="calculator-links">
          <li className="calculator-links-li">
            <b>Addition: </b>
            Adding items together.
          </li>
          <li className="calculator-links-li">
            <b>Subtraction: </b>
            Subtracting things together.
          </li>
          <li className="calculator-links-li">
            <b>Multiplication: </b>
            Multiplying things together.
          </li>
          <li className="calculator-links-li">
            <b>Division: </b>
            Divide things togther.
          </li>
          <li className="calculator-links-li">
            <b>Modulus: </b>
            Multiply things together.
          </li>
        </ul>
      </div>
      <Calculator />
    </div>
  );
}

export default CalculatorPage;
