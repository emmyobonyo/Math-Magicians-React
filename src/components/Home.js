import React from 'react';

function Home() {
  return (
    <div className="home-div">
      <h2>Welcome to our page</h2>
      <p>
        Hello there, glad you stopped by.
        We appreciate the fact that you have a
        calculator on your phone -- probably a
        better one -- but you decided to spend
        some data to check out our site.
        Or Maybe your phone just run out of battery?
        Either way we are glad that you came and hopefully
        your phone runs out of battery more often.
      </p>
      <p>
        So to the basics. You can perfom basic mathematical
        calculations (Additions, Divisions, Multiplication,
        Modulus, and Subtraction). What about the other
        Arithmetic operations? You may have to buy an actual
        calculator for that.
      </p>
      <p>Thank you for taking the time to visit and use my calculator</p>
    </div>
  );
}

export default Home;
