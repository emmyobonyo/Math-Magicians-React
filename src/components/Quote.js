import React from 'react';

function Quote() {
  return (
    <div className="quote-page">
      <h2>
        Programming is not about numbers,
        data structures, variables, classes,
        debugging, git-flow, cramming,
        copying and pasting, stack-overflow
        or even functions. It is about understanding.
      </h2>
    </div>
  );
}

export default Quote;
