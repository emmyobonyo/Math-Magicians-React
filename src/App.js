import React from 'react';
import Calculator from './components/calculator';
import './App.css';
// import Calculate from './logic/calculate';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = { total: 0 };
  }

  render() {
    const { total } = this.state;
    return (
      <div>
        <Calculator total={total} />
      </div>
    );
  }
}

export default App;
