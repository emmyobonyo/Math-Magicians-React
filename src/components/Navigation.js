import React from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link,
} from 'react-router-dom';
import Calculator from './calculator';
import Quote from './Quote';
import Home from './Home';

function Navigation() {
  return (
    <Router>
      <div>
        <nav className="navigation">
          <h3 className="logo">
            <Link to="/">Math Magicians</Link>
          </h3>
          <ul className="navigation-links">
            <li>
              <Link to="/">Home</Link>
            </li>
            <li className="link-borders">
              <Link to="/calculator">Calculator</Link>
            </li>
            <li>
              <Link to="/quote">Quote</Link>
            </li>
          </ul>
        </nav>
        <Switch>
          <Route path="/calculator">
            <Calculator />
          </Route>
          <Route path="/quote">
            <Quote />
          </Route>
          <Route path="/">
            <Home />
          </Route>
        </Switch>
      </div>
    </Router>
  );
}

export default Navigation;
