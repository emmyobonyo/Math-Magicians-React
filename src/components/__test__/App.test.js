import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from '../../App';
import CalculatorPage from '../CalculatorPage';
import Navigation from '../Navigation';
import Home from '../Home';
import Quote from '../Quote';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('matches snapshot', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<CalculatorPage />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('matches snapshot', () => {
  const tree = renderer.create(<CalculatorPage />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Navigation />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('matches snapshot', () => {
  const tree = renderer.create(<Navigation />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Home />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('matches snapshot', () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Quote />, div);
  ReactDOM.unmountComponentAtNode(div);
});
