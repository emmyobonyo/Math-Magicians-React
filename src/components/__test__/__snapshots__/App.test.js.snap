// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div>
    <nav
      className="navigation"
    >
      <h3
        className="logo"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Math Magicians
        </a>
      </h3>
      <ul
        className="navigation-links"
      >
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="link-borders"
        >
          <a
            href="/calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <li>
          <a
            href="/quote"
            onClick={[Function]}
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="home-div"
    >
      <h2>
        Welcome to our page
      </h2>
      <p>
        Hello there, glad you stopped by. We appreciate the fact that you have a calculator on your phone -- probably a better one -- but you decided to spend some data to check out our site. Or Maybe your phone just run out of battery? Either way we are glad that you came and hopefully your phone runs out of battery more often.
      </p>
      <p>
        So to the basics. You can perfom basic mathematical calculations (Additions, Divisions, Multiplication, Modulus, and Subtraction). What about the other Arithmetic operations? You may have to buy an actual calculator for that.
      </p>
    </div>
  </div>
</div>
`;

exports[`matches snapshot 2`] = `
<div
  className="calculator-page"
>
  <div>
    <h2
      className="calculator-page-heading"
    >
      Let us do some Math
    </h2>
    <p>
      These are the calculations that you can perform
    </p>
    <ul
      className="calculator-links"
    >
      <li
        className="calculator-links-li"
      >
        <b>
          Addition: 
        </b>
        Adding items together.
      </li>
      <li
        className="calculator-links-li"
      >
        <b>
          Subtraction: 
        </b>
        Subtracting things together.
      </li>
      <li
        className="calculator-links-li"
      >
        <b>
          Multiplication: 
        </b>
        Multiplying things together.
      </li>
      <li
        className="calculator-links-li"
      >
        <b>
          Division: 
        </b>
        Divide things togther.
      </li>
      <li
        className="calculator-links-li"
      >
        <b>
          Modulus: 
        </b>
        Multiply things together.
      </li>
    </ul>
  </div>
  <div
    className="calculator"
  >
    <div
      className="calculator-screen"
    >
      <p
        className="calculator-digits"
      >
        0  
      </p>
    </div>
    <div
      className="calculator-buttons"
    >
      <button
        className="calculator-button"
        data-testid="ac"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="calculator-button"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="calculator-button"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="calculator-button orange-button"
        onClick={[Function]}
        type="button"
      >
        /
      </button>
      <button
        className="calculator-button"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="calculator-button"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="calculator-button"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="calculator-button orange-button"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="calculator-button"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="calculator-button"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="calculator-button"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="calculator-button orange-button"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="calculator-button"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="calculator-button"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="calculator-button"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="calculator-button orange-button"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="calculator-buttons-bottom"
    >
      <button
        className="calculator-button-bottom button-one"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="calculator-button-bottom button-two"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="calculator-button-bottom button-three orange-button-bottom"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`matches snapshot 3`] = `
<div>
  <nav
    className="navigation"
  >
    <h3
      className="logo"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Math Magicians
      </a>
    </h3>
    <ul
      className="navigation-links"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="link-borders"
      >
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="home-div"
  >
    <h2>
      Welcome to our page
    </h2>
    <p>
      Hello there, glad you stopped by. We appreciate the fact that you have a calculator on your phone -- probably a better one -- but you decided to spend some data to check out our site. Or Maybe your phone just run out of battery? Either way we are glad that you came and hopefully your phone runs out of battery more often.
    </p>
    <p>
      So to the basics. You can perfom basic mathematical calculations (Additions, Divisions, Multiplication, Modulus, and Subtraction). What about the other Arithmetic operations? You may have to buy an actual calculator for that.
    </p>
  </div>
</div>
`;

exports[`matches snapshot 4`] = `
<div
  className="home-div"
>
  <h2>
    Welcome to our page
  </h2>
  <p>
    Hello there, glad you stopped by. We appreciate the fact that you have a calculator on your phone -- probably a better one -- but you decided to spend some data to check out our site. Or Maybe your phone just run out of battery? Either way we are glad that you came and hopefully your phone runs out of battery more often.
  </p>
  <p>
    So to the basics. You can perfom basic mathematical calculations (Additions, Divisions, Multiplication, Modulus, and Subtraction). What about the other Arithmetic operations? You may have to buy an actual calculator for that.
  </p>
</div>
`;
